File: 242-valid-anagram/valid-anagram.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isAnagram = function(s, t) {

    if(s.length!==t.length) return false;

    let sMap = {};
    let tMap = {}

    for(let i=0;i<s.length;i++) {

        if(!sMap[s[i]]) sMap[s[i]]=1;
        else sMap[s[i]]++;

        if(!tMap[t[i]]) tMap[t[i]]=1;
        else tMap[t[i]]++
    }

    let keyArr = Object.keys(sMap);
    for(let i=0;i<keyArr.length;i++) {

        if(sMap[keyArr[i]]!==tMap[keyArr[i]]) return false;
    }
    return true;


    
};

File: 242-valid-anagram/valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        
        if(s.length()!=t.length()) return false;

        Map<Character,Integer> sMap = new HashMap<>();
        Map<Character,Integer> tMap = new HashMap<>();

        for(int i=0;i<s.length();i++) {

            char sChar = s.charAt(i);
            char tChar = t.charAt(i);

            if(sMap.get(sChar)==null) sMap.put(sChar,1);
            else sMap.put(sChar,sMap.get(sChar)+1);

             if(tMap.get(tChar)==null) tMap.put(tChar,1);
            else tMap.put(tChar,tMap.get(tChar)+1);
        }

        Object[] keyArr = sMap.keySet().toArray();

        for(int i=0;i<keyArr.length;i++) {

            if(!sMap.get(keyArr[i]).equals(tMap.get(keyArr[i]))) return false;
        
        }

        return true;


    
    }
}